### PR TITLE
Feature: Delete container api in group single page

### PR DESCRIPTION
**Pull request by ShoeheyOt**
## Overview

Implement calling API for delete container including validation/schema

## Changes

 some file change only includes adding props to pass, I left a comment in some files which do so and you can ignore.

## Screen Captures

[DeleteContainer_API.webm](https://github.com/nishiki-tech/nishiki-frontend/assets/142355969/12acd0f2-416e-4ce7-9bf9-bdb57beb948d)

## Assignee Checklist:

- [X] The naming convention of the PR title is correct (See the comment at the top of this template)
- [X] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [X] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [X] The correct assignees and reviewers have been designated for this PR
- [X] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [X] All the related issues are associated with this PR
- [X] All criteria in the associated issues are met (please tick the checkboxes)
- [X] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/DeleteContainerDialogTriggerButton/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/DeleteContainerDialogTriggerButton/index.tsx:1]: The change of this file is only passing `containerId` from parent to child component, please ignore this file

File: src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/ContainerCardDropdownMenuContent/index.tsx:1]: The change of this file is also only passing `containerId`, please ignore this file too.

File: src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/ContainerCardList/ContainerCard/ContainerCardDropdownMenuTriggerButton/index.tsx:1]: The change of this file is also only passing `containerId`, please ignore this file too.

File: src/features/groups/components/ContainerCardList/ContainerCard/index.tsx
Comment (ShoeheyOt) [inline src/features/groups/components/ContainerCardList/ContainerCard/index.tsx:1]: The change of this file is also only passing `containerId`, please ignore this file too.

File: src/lib/api/container/client/containerApiClient.client.test.ts [hunk 1]
```diff
@@ -94,6 +96,35 @@ describe('containerApiClient', () => {
...
+      it('should return Err result if API request fails', async () => {
+        /* Arrange */
+        const mockError = new Error('API error');
+        (request as jest.Mock).mockRejectedValue(mockError);
+
+        /* Act */
+        const result = await deleteContainer(mockContainerId);
+
+        /* Assert */
+        expect(result.unwrapError()).toBe(mockError.message);
```
Comment (nick-y-ito) [inline src/lib/api/container/client/containerApiClient.client.test.ts:125]: Adding `expect(result.err).toBeTruthy();` will improve the clarity of this assertion.

```ts
expect(result.err).toBeTruthy();
expect(result.unwrapError()).toBe(mockError.message);
```

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@ShoeheyOt 
Thank you for your work! It is working well!

Please take a look at my comments and consider addressing them.

File: src/lib/api/container/client/containerApiClient.client.test.ts [hunk 2]
```diff
@@ -94,6 +96,35 @@ describe('containerApiClient', () => {
...
+    describe('deleteContainer', () => {
+      it('should return Ok result on success', async () => {
+        /* Arrange */
+        (request as jest.Mock).mockResolvedValue({});
+
+        /* Act */
+        const result = await deleteContainer(mockContainerId);
+
+        /* Assert */
+        expect(result.unwrap()).toBe(undefined);
```
[the remark this answers is not in the document]
Comment (ShoeheyOt) [reply src/lib/api/container/client/containerApiClient.client.test.ts]: @nick-y-ito  Thank you for your review, I understand your point, I assume I need to update not only here but also in `actions.test` which I have done already.

**Review by nick-y-ito — APPROVED** [review]
@ShoeheyOt
Thank you for the update! I think now this PR is ready for merging!

Regarding the testing, let's stick to this notation from next time when we come across up-updated parts.
```ts
expect(result.ok).toBeTruthy();
expect(result.unwrap).tobe(ExpectedValue);

expect(result.err).toBeTruthy();
expect(result.unwrapError()).toBe('Error Message');
```

**Comment by ShoeheyOt** [comment]
@nick-y-ito Thank you for re-reviewing! yes, let's stick to the notation you suggested!

**Pull request merged**